enclave/core/CMakeLists.txt

```diff
@@ -275,7 +275,11 @@ if (OE_SGX)
     # user. There are valid reasons for an end user to use built-ins.
     $<BUILD_INTERFACE:-fno-builtin-malloc
     -fno-builtin-calloc
-    -fno-builtin>)
+    -fno-builtin>
+    # Treat warnings as errors by default. Application based on OE SDK
```

Comment:
Is this desirable? This means that every application that uses liboecore via CMake import or pkgconfig will get `-Werror` by default, not just the samples. It's not generally my experience that SDKs set this on behalf of a dev (folks should feel to correct this if I am mistaken.)

This is also potentially a public build breaking change and should at least be documented in the CHANGELOG.md.

Reply:
Good point. I'm unapproving till we are sure that we don't force these flags on the users.

Reply:
This PR is based on the conclusion from #3773. I'm fine to remove the enforcement o Windows side. Can we have a consensus? Cc @radhikaj @dthaler

Reply:
There is a distinction:
- We want the samples to be built using -Werror
- However, we don't want to force -Werror on the user.

My understanding is that, today, we don't force the flags on the user on both Linux and Windows.

Reply:
No. -Werror is added in the clangw by default, so samples on Windows are always compiled with the flag, which is what the #3773 about.

Reply:
Thanks for the clarification. We were forcing the flag on Windows, whereas we weren't on Linux.
I'm removing my blocker. 

It is a matter of consistency then.

Reply:
Exactly. Given that @CodeMonkeyLeet pointed out that we shouldn't enforce -Werror on behalf of users, I'm wondering how should we address the consistency? This PR currently follows the conclusion from triaging #3733.  We could also remove the enforcement on the Windows side. Thoughts?

Reply:
There are convincing arguments for both choices. Thinking about it bit more, my preference would be to enforce -Werror since it is generally considered a good programming practice to compile code with high warning levels.

Reply:
+1 to Anand's summary before about what is desired state:
* We want the samples to be built using -Werror
* However, we don't want to force -Werror on the user.

Above should be true for both Linux and Windows.